Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/TaskRecyclerViewAdapter.java

```diff
@@ -0,0 +1,124 @@
+package pg.autyzm.friendly_plans;
+
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import database.entities.TaskTemplate;
+
+
+/**
+ * Created by Mateusz on 2017-03-08.
+ */
+
+public class TaskRecyclerViewAdapter extends RecyclerView.Adapter<TaskRecyclerViewAdapter.TaskListViewHolder> {
+
+
+    private List<TaskTemplate> taskItemList;
+    private TaskItemClickListener taskItemClickListener;
+    private int ICON_PLACEHOLDER_ID = R.drawable.ic_placeholder;
+
+    @Override
+    public TaskListViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_task, parent, false);
+        return new TaskListViewHolder(view, taskItemClickListener);
+    }
+
+    @Override
+    public void onBindViewHolder(TaskListViewHolder holder, int position) {
+
+        if (taskItemList != null || taskItemList.size() != 0) {
+
+            TaskTemplate taskItem = taskItemList.get(position);
+            holder.taskName.setText(taskItem.getName());
+            //condition only for mocking purposes -> needs to be changed later
+            if (taskItem.getPicture()!= null && !taskItem.getPicture().isEmpty()) {
+                holder.taskPicture.setImageResource(ICON_PLACEHOLDER_ID);
+                holder.taskPicture.setVisibility(View.VISIBLE);
+            } else  {
+                holder.taskPicture.setVisibility(View.GONE);
+            }
+            if (taskItem.getSound()!= null && !taskItem.getSound().isEmpty()) {
+                holder.taskSoundIcon.setImageResource(ICON_PLACEHOLDER_ID);
+                holder.taskSoundIcon.setVisibility(View.VISIBLE);
+            } else  {
+                holder.taskSoundIcon.setVisibility(View.GONE);
+            }
+            if (taskItem.getDurationTime() != 0) {
+                holder.taskDurationIcon.setImageResource(ICON_PLACEHOLDER_ID);
+                holder.taskDurationTime.setText(String.valueOf(taskItem.getDurationTime()));
+                holder.taskDurationIcon.setVisibility(View.VISIBLE);
+            } else {
+                holder.taskDurationTime.setText("");
+                holder.taskDurationIcon.setVisibility(View.INVISIBLE);
+            }
+        }
+
+
+    }
+
+
+    @Override
+    public int getItemCount() {
+        return ((taskItemList != null) && (taskItemList.size() != 0) ? taskItemList.size() : 0);
+    }
+
+
+    public interface TaskItemClickListener {
```

Comment:
I do some quick research about this, because it looks a little bit confusing for me, but for now I don't have any idea how connect Handler and Activity in a different way. What do you think about this construction in general? For me it's a little bit odd to pass the listener this way but maybe I'm just not used to it :D
What do you think about moving interface to another file? Maybe there is no other way? Or maybe this approach is just right? :)